crypto/internal.h

```diff
@@ -272,8 +276,44 @@ static inline uint32_t CRYPTO_bswap4(uint32_t x) {
 static inline uint32_t CRYPTO_bswap4(uint32_t x) {
   return _byteswap_ulong(x);
 }
+
+static inline uint64_t CRYPTO_bswap8(uint64_t x) {
+  return _byteswap_uint64(x);
+}
+#else
+static inline uint32_t CRYPTO_bswap4(uint32_t x) {
+  x = (x >> 16) | (x << 16);
+  x = ((x & 0xff00ff00) >> 8) | ((x & 0x00ff00ff) << 8);
+  return x;
+}
+
+static inline uint64_t CRYPTO_bswap8(uint64_t x) {
+  return CRYPTO_bswap4(x >> 32) | (((uint64_t)CRYPTO_bswap4(x)) << 32);
+}
 #endif
 
+
+// Convert LE or BE values to/from native endianness.
+
+#if defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__)
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#define RING_BIG_ENDIAN
+#endif
+#endif
+
+#ifdef RING_BIG_ENDIAN
+#define CRYPTO_bswap4_le(x) CRYPTO_bswap4((x))
+#define CRYPTO_bswap8_le(x) CRYPTO_bswap8((x))
+#define CRYPTO_bswap4_be(x) (x)
+#define CRYPTO_bswap8_be(x) (x)
+#else
+#define CRYPTO_bswap4_le(x) (x)
+#define CRYPTO_bswap8_le(x) (x)
+#define CRYPTO_bswap4_be(x) CRYPTO_bswap4((x))
+#define CRYPTO_bswap8_be(x) CRYPTO_bswap8((x))
+#endif
```

Comment:
In general we don't use `#define` for functions; instead we use `static inline` functions. To help you, I brought in the BoringSSL definitions of the following functions in the recent BoringSSL merge:
```
static inline uint32_t CRYPTO_load_u32_le(const void *in);
static inline void CRYPTO_store_u32_le(void *out, uint32_t v);
static inline uint32_t CRYPTO_load_u32_be(const void *in);
static inline void CRYPTO_store_u32_be(void *out, uint32_t v);
static inline uint64_t CRYPTO_load_u64_le(const void *in);
static inline void CRYPTO_store_u64_le(void *out, uint64_t v);
static inline uint64_t CRYPTO_load_u64_be(const void *ptr);
static inline void CRYPTO_store_u64_be(void *out, uint64_t v);
static inline crypto_word_t CRYPTO_load_word_le(const void *in);
static inline void CRYPTO_store_word_le(void *out, crypto_word_t v);
static inline crypto_word_t CRYPTO_load_word_be(const void *in);
```
Please use these functions throughout the PR as they are more consistent with the Rust equivalents.

Basically I think your PR will mostly reduce to implementing the big-endian versions of these functions.

Note that some of these functions may be dead code in *ring* now and you may just remove the dead ones instead of porting them to big-endian, if you want.

Reply:
I've removed the load_word / store_word versions as they are unneded.  Everything else I've added big-endian versions for.